File: src/python/leetcode-problem-1.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        data = dict()
        for i, v in enumerate(nums):
            if target - v in data:
                return [data[target - v], i]
            data[v] = i

print(Solution().twoSum([2, 7, 11, 15], 9))

File: src/java/leetcode-problem-1.java
import java.util.Arrays;
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashtable = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (hashtable.containsKey(target - nums[i])) {
                return new int[] {hashtable.get(target - nums[i]), i};
            }
            hashtable.put(nums[i], i);
        }
        return null;
    }

    public static void main(String args[]) {
        Solution s = new Solution();
        System.out.println(Arrays.toString(s.twoSum(new int[] {2, 7, 9, 11}, 9)));
        System.out.println(Arrays.toString(s.twoSum(new int[] {3, 2, 4}, 6)));
    }
}
